letsencrypt/cli.py

```diff
@@ -198,81 +199,79 @@ def _find_duplicative_certs(domains, config, renew_config):
-            ).format(identical_names_cert.configfile.filename)
+            ).format(ident_names_cert.configfile.filename, br=os.linesep)
         elif subset_names_cert is not None:
             question = (
                 "You have an existing certificate that contains a portion of "
-                "the domains you requested (ref: {0})\n\nIt contains these "
-                "names: {1}\n\nYou requested these names for the new "
-                "certificate: {2}.\n\nDo you want to replace this existing "
+                "the domains you requested (ref: {0}){br}{br}It contains these "
+                "names: {1}{br}{br}You requested these names for the new "
+                "certificate: {2}.{br}{br}Do you want to replace this existing "
                 "certificate with the new certificate?"
             ).format(subset_names_cert.configfile.filename,
                      ", ".join(subset_names_cert.names()),
-                     ", ".join(domains))
+                     ", ".join(domains),
+                     br=os.linesep)
         if question is None:
             # We aren't in a duplicative-names situation at all, so we don't
             # have to tell or ask the user anything about this.
             pass
-        elif zope.component.getUtility(interfaces.IDisplay).yesno(
-                question, "Replace", "Cancel"):
-            treat_as_renewal = True
+        elif config.no_confirm or zope.component.getUtility(
+                interfaces.IDisplay).yesno(question, "Replace", "Cancel"):
+            renewal = True
         else:
             reporter_util = zope.component.getUtility(interfaces.IReporter)
             reporter_util.add_message(
                 "To obtain a new certificate that {0} an existing certificate "
                 "in its domain-name coverage, you must use the --duplicate "
-                "option.\n\nFor example:\n\n{1} --duplicate {2}".format(
-                    "duplicates" if identical_names_cert is not None else
-                    "overlaps with", sys.argv[0], " ".join(sys.argv[1:])),
+                "option.{br}{br}For example:{br}{br}{1} --duplicate {2}".format(
+                    "duplicates" if ident_names_cert is not None else
+                    "overlaps with",
+                    sys.argv[0], " ".join(sys.argv[1:]),
+                    br=os.linesep
+                ),
                 reporter_util.HIGH_PRIORITY)
-            return 1
+            raise errors.Error(
+                "User did not use proper CLI and would like "
+                "to reinvoke the client.")
 
-    # Attempting to obtain the certificate
-    # TODO: Handle errors from _init_le_client?
-    le_client = _init_le_client(args, config, authenticator, installer)
-    if treat_as_renewal:
-        lineage = identical_names_cert if identical_names_cert is not None else subset_names_cert
-        # TODO: Use existing privkey instead of generating a new one
```

Comment:
The biggest use-case I can think of for keeping the same key is to be able to use the [HTTP Public-Key-Pins header (HPKP)](https://tools.ietf.org/html/rfc7469).

With HPKP, a server needs to have at least 2 public keys listed: one for the current certificate, and at least one backup key that should not be stored on the server. I would want to continue using the currently active key to ensure that someone who visits my site and sees my HPKP header just before I renew my certificate under normal circumstances does not get locked out of my site for any length of time. If the key rotates all the time (Even if every 60-90 days), then there wouldn't be a safe way to use HPKP with lets encrypt (The suggested length of time for the HPKP max-age is 60 days, which is close to how often a server operator would need to rotate their lets encrypt certificate).